addons/purchase/models/account_invoice.py

```diff
@@ -36,7 +36,7 @@ def _prepare_invoice_line_from_po_line(self, line):
         else:
             qty = line.qty_received - line.qty_invoiced
         if float_compare(qty, 0.0, precision_rounding=line.product_uom.rounding) <= 0:
-            qty = 0.0
+            qty = qty
```

Comment:
Ok, so this actually needs to be 
```python
if float_compare(qty, 0.0, precision_rounding=line.product_uom.rounding) == 0:
     qty = 0.0
```
I don't know why the rounding was needed but apparently it was (otherwise we'd just have `if qty < 0:`).

Apologies if I offended anyone with my first comment. It wasn't my intention.